wordpress/wp-content/plugins/cds-base/classes/Modules/Cleanup/AdminBar.php

```diff
@@ -12,6 +12,113 @@ public function __construct()
         add_action('wp_before_admin_bar_render', [$this, 'removeFromAdminBarBefore'], 99);
 
         add_action('admin_bar_menu', [$this, 'addCollections'], 21);
+        add_action('admin_bar_menu', [$this, 'addAdminToggle'], 21);
+
+        add_action('admin_bar_menu', [$this, 'addLanguageSwitcher'], 21);
+        add_action('admin_post_cds_change_lang', [$this, 'handleLanguageSwitcherResponse']);
+    }
+
+    public function addAdminToggle($wp_admin_bar): void
+    {
+        $menu_id = 'cds-home';
+        // "is_admin" checks if viewing an admin page, it's not a role check
+        $home_url = is_admin() ? home_url('/') : admin_url();
+
+        $wp_admin_bar->add_node([
+            'id'    => $menu_id,
+            'title' => '<div class="ab-item"><span class="ab-icon"></span>' . __(
+                'GC Articles:',
+                'cds-snc'
+            ) . ' ' . get_bloginfo('name') . '</div>',
+            'href'  => $home_url,
+        ]);
+
+        // Add an option to visit the site.
+        $wp_admin_bar->add_node(
+            array(
+                'parent' => $menu_id,
+                'id'     => $menu_id . '-view-site',
+                'title'  => __('Visit Site'),
+                'href'   => home_url('/'),
+            )
+        );
+
+        // Add an option to visit the site's "settings" page for superadmins
+        if (is_multisite() && current_user_can('manage_sites')) {
+            $wp_admin_bar->add_node(
+                array(
+                    'parent' => $menu_id,
+                    'id'     => $menu_id . '-edit-site',
+                    'title'  => __('Edit Site'),
+                    'href'   => network_admin_url('site-info.php?id=' . get_current_blog_id()),
+                )
+            );
+        }
+
+        // Add an option to visit the dashboard.
+        $wp_admin_bar->add_node(
+            array(
+                'parent' => $menu_id,
+                'id'     => $menu_id . '-admin-dashboard',
+                'title'  => __('Admin dashboard'),
+                'href'   => admin_url(),
+            )
+        );
+    }
+
+    public function addLanguageSwitcher($wp_admin_bar): void
+    {
+        $user_locale = get_user_locale();
+        $translation_locale = str_contains($user_locale, "en") ? "fr_FR" : "en_US";
+        // Not using i18n function for these because we don't want to translate them
+        $translation_locale_name = str_contains($user_locale, "en") ? "Français" : "English";
```

Comment:
The language strings are hardcoded here, so probably it's better to move them out somewhere.